### PR TITLE
feat: ✨ drop-down

### LINKED ISSUE #315: feat: ✨ drop-down
# Description
The drop-down is needed for the header. It is used  as a menu if a header icon is clicked.

# Acceptance criteria

- [ ] All DoD items successfully completed
- [ ] Incorporates slots, Events, Methods, Parts from Shoelace
- [ ] Tested: Playwright + Chromatic
- [ ] create a proper documentation which contains at least everything from the Shoelace component
- [ ] Text in the documentation can be synced

### PR DESCRIPTION
**Pull request by LeaLeRose**
# Pull Request

## 📖 Description

This PR provides components, tests, stories and documentation that are needed to use `<syn-dropdown />`. It provides the following new components, as well as their framework wrappers:

- `<syn-dropdown />`, which is used to display dropdowns
- `<syn-menu />`, which may be used to structure the items in a `<syn-dropdown />`
- `<syn-menu-item />`, which renders a an option in a `<syn-menu />`
- `<syn-menu-label />`, that allows to group multiple `<syn-menu-item />`s together
- `<syn-popup />`, which is a low level utility not intended for direct use that positions the `<syn-dropdown />`

### 🎫 Issues

Closes #315

## 👩‍💻 Reviewer Notes

- Please have a look at the corresponding Storybook stories
- We decided to postphone the shipping of our custom `<syn-nupsi />`, as this is too much work for the time being. Ticket status of the `<syn-nupsi />` implementation is tracked in #366 for dev.
- ⚠️ The `<syn-menu />` provides a hover state that seems to be sticky when leaving the component. This is not the case when visiting shoelaces demo, but also the case when creating new codepens that use `<sl-menu />`. This has to be investigated. However, it does not seem to be a problem when using the `<syn-menu />` in a `<syn-dropdown />`, which should be the case almost everywhere.

## 📑 Test Plan

For Devs: Just follow the regular procedure.

## ⚠️ Known issues:

There is currently an issue in Shoelace with overlapping `:focus` and `:hover` styles. The `:focus` pseudo state is always triggered and has preference. Therefore, we will always get a blue background color when hovering or activating an item, unless we move down a submenu where the light-grey color is used for the parent. This will not be fixed on our side as shoelace is currently working on this issue for their 3.0 milestone. See https://github.com/shoelace-style/shoelace/issues/1676 and and https://github.com/shoelace-style/shoelace/issues/1688 for further information.

## ✅ DoD

- [x] I have tested my changes manually.
- [x] I have added automatic tests for my changes (unit- and visual regression tests).
- [x] I have updated the project documentation to reflect my changes (e.g. CHANGELOG, Storybook, ...).
- [x] I have made sure to follow the projects coding and contribution guides.
- [ ] ~~I have made sure that the bundle size has changed appropriately.~~
- [x] I have validated that there are no accessibility errors.
- [x] I have used design tokens instead of fix css values

### CODE REVIEW

**Comment by kirchsuSICKAG** [comment]
In react the dropdown does not have the same behaviour, that the webcomponent and the other frameworks. The dropdown does not auto-close if an item is clicked, but it stays open.

Used code for testing: 

```
        <SynDropdown skidding={30} >
          <SynButton slot="trigger" caret>Edit</SynButton>
          <SynMenu>
            <SynOption>Cut</SynOption>
            <SynOption>Copy</SynOption>
            <SynOption>Paste</SynOption>
            <SynDivider />
            <SynOption>Find</SynOption>
            <SynOption>Replace</SynOption>
          </SynMenu>
        </SynDropdown>
```

**Comment by schilchSICKAG** [comment]
> In react the dropdown does not have the same behaviour, that the webcomponent and the other frameworks. The dropdown does not auto-close if an item is clicked, but it stays open.
> 
> Used code for testing:
> 
> ```
>         <SynDropdown skidding={30} >
>           <SynButton slot="trigger" caret>Edit</SynButton>
>           <SynMenu>
>             <SynOption>Cut</SynOption>
>             <SynOption>Copy</SynOption>
>             <SynOption>Paste</SynOption>
>             <SynDivider />
>             <SynOption>Find</SynOption>
>             <SynOption>Replace</SynOption>
>           </SynMenu>
>         </SynDropdown>
> ```

Maybe this is because you have to use `<syn-menu-item />`and not `<syn-option />` here.

**Review by kirchsuSICKAG — CHANGES_REQUESTED** [review]

**Comment by schilchSICKAG** [comment]
> > Maybe this is because you have to use `<syn-menu-item />`and not `<syn-option />` here.
> 
> You are completely right. I trusted copilot there and didn`t check xD

I was afraid and had a manual look. In my case the background was transparent. Turned out my CoPilot was not adding `<SynMenu />` around the items. Seems like it has a lot to learn :)

File: packages/components/src/components/menu-item/menu-item.custom.styles.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import { css } from 'lit';
+
+export default css`
+  .menu-item {
+    color: var(--syn-typography-color-text);
+    font: var(--syn-font-size-medium);
```
[the remark this answers is not in the document]
Comment (schilchSICKAG) [reply packages/components/src/components/menu-item/menu-item.custom.styles.ts:6]: Not at all. It works, but I also think `font-size` makes the intent more clear. Updated.

File: packages/components/src/themes/utility.css [hunk 2]
```diff
@@ -4,9 +4,18 @@
...
+  .syn-scroll-lock {
+    overflow: hidden !important;
+    scrollbar-gutter: stable !important;
+  }
+}
+
+@supports not (scrollbar-gutter: stable) {
+  .syn-scroll-lock {
+    overflow: hidden !important;
+    padding-right: var(--sl-scroll-lock-size) !important;
```
[the remark this answers is not in the document]
Comment (schilchSICKAG) [reply packages/components/src/themes/utility.css:17]: Good catch!

File: packages/docs/stories/components/dropdown.stories.ts [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+        story: generateStoryDescription('dropdown', 'default'),
+      },
+    },
+  },
+  render: (args: unknown) => html`
+    <div style="position: relative">
+      ${generateTemplate({ args })}
+    </div>
+  `,
+} as Story;
```
[the remark this answers is not in the document]
Comment (schilchSICKAG) [reply packages/docs/stories/components/dropdown.stories.ts:98]: There is unfortunately nothing we could do against this (internal implementation detail of shoelace)

**Review by kirchsuSICKAG — APPROVED** [review]

**Pull request merged**